### ISSUE TITLE
Better Error Messaging when PID provider is unavailable

### ISSUE THREAD

**Issue by djbrooke** [IQSS/dataverse#4819]
Moving the PR that's currently under #4800 here instead. We'll move this through the process.

From @qqmyers: 

when DataCite/the PID provider is down (non-201 response), the original code was just ignoring the issue. This change triggers the configured error message from Bundle.properties that occurs if an exception is thrown from the service call.
Also updated error to end with 'if the problem persists, please contact <contact.support>' rather than 'Does issue continue to persist?'

**Comment by mheppler** [comment]
@qqmyers:

I had this issue pulled from QA in order to review bundle change in the PR, in order to see what it would look like on the page. While the rest of the PR adds much needed details and context to the error msg in the screenshot from the original comment by @pdurbin in issue #4800, I believe that the text edit to the bundle message is redundant because of how it gets delivered to the UI.

The `FinalizeDatasetPublicationCommand.java` passes the `dataset.publish.error` bundle message to the UI through the UI component in `messages.xhtml`, where it gets combined the following  byndle message:

```
error.support.message= If you believe this is an error, please contact {0} for assistance.
```

If we revert the bundle edit back, this is how the message should read in the UI, with both bundle messages combined:

![screen shot 2018-07-10 at 2 13 13 pm](https://user-images.githubusercontent.com/687227/42530132-643e6b08-844e-11e8-9499-32a4bef04765.png)

The support team name and link are already in the UI for this and all error messages, so there is no need to add it again.

**Comment by qqmyers** [comment]
I agree the contact is redundant. However, I think the 'Does the issue continue to persist?' is not very grammatical and perhaps unnecessary. I left it off in my commit. I won't be able to make further changes in the near future, so if there's better language or you want to add this phrase back, go ahead - I think you should be able to edit the PR//branch.

**Comment by mheppler** [comment]
Thank you for the revision. This is something that we will review when we pick up the _Style guide – document messaging text and visual styles #3865_ issue.

I will pass this issue back along to QA.

**Comment by pdurbin** [comment]
Pull request #4606 was just merged and now pull request #4807 has merge conflicts that should be resolved before this is QA'ed.

Also, I thought of an idea of how to maybe test this issue while DataCite is up. We could retest #3328 by putting an `&nbsp;` in a dataset description to see if we get a better error for the user than "PublishDatasetCommand@2e107f1d failed: null" or whatever.

**Comment by pdurbin** [comment]
Nope, as of 90daeb5 at least, you still get an ugly "Error – Command edu.harvard.iq.dataverse.engine.command.impl.FinalizeDatasetPublicationCommand@56ebb36a failed: null" like this if you put an `&nbsp;` in a dataset description:

![screen shot 2018-07-13 at 4 33 38 pm](https://user-images.githubusercontent.com/21006/42712818-18c1cbec-86bb-11e8-9a9d-4b5a4e6b24a7.png)

![screen shot 2018-07-13 at 4 38 11 pm](https://user-images.githubusercontent.com/21006/42712833-251d1306-86bb-11e8-9560-4458255b8689.png)

@qqmyers please don't take the above as a criticism! I only jumped on this branch to see if I should try to resolve the merge conflicts but I have to head out. It might be nice to handle this error better too though, if it's related to the code in the pull request. Up to you if you want to try to take this on.

Again, someone should resolve the merge conflicts before the pull request is tested. Here's the conflict:

![screen shot 2018-07-13 at 4 43 55 pm](https://user-images.githubusercontent.com/21006/42713055-069027f6-86bc-11e8-9bab-282f5e5f809b.png)

**Comment by pdurbin** [comment]
I pulled this issue out of QA because the merged conflicts above need to be resolved before testing.

**Comment by qqmyers** [comment]
@pdurbin - I've addressed the conflict, but #3328 exposes a fun wrinkle. Apparently, the 400 response code from having &nbsp; in the description causes a transaction rollback which then causes an exception in the catch clause as part of the getProviderInformation() call. (Looks like the idServiceBean is null due to the earlier rollback.). I'm not sure I understand EJB enough to explain this, but calling getProviderInformation() before the exception as in this PR (just in case its needed) works and allows the dataset.publish.error to be displayed.

I verified this for #3328 and expect it will be the same for any non-201 response or IOException since they are all handled with the same code resulting in a RuntimeException. That is, I think the last commit is needed for the case when DataCite is down.

The remaining issue w.r.t. #3328 is that, while this error is handled by this PR, it results in the dataset.publish.error message which was defaulting to "This dataset may not be published because the <a href=\{1} target=\"_blank\"/> {0} </a> Service  is currently inaccessible. Please try again."
I've updated that to "This dataset may not be published due to an error when contacting the <a href=\{1} target=\"_blank\"/> {0} </a> Service. Please try again." which may be better than before but might still need tweaking (i.e. trying again won't help with #3328).
